### PR TITLE
Fix generate:types bug when slug has underscore prefix

### LINKED ISSUE #3697: generate:types reference incorrect type, if collection slug has _underline prefix
### Link to reproduction

https://github.com/xHomu/payload-generate-types-error

### To Reproduce

1. yarn generate:types
2. type error, orders type should be _Order, not _order

image

### Describe the Bug

This occurs because the new `generateEntityDeclarations()` function generates TypeName that doesn't match those generated by `json-schema-to-typescript`.

I believe this can be solved by referencing instead to the provided Config type, ie:

```jsx
declare module "payload" {
  export interface GeneratedTypes extends Config {} 
}
```

### Payload Version

2.0.7

### Adapters and Plugins

db-mongodb

### PR DESCRIPTION
**Pull request by xHomu**
## Description

generateEntityDeclarations function creates mismatched type names. We'll simply use the existing Config type instead.

Closes #3697 

- [x] I have read and understand the [CONTRIBUTING.md](https://github.com/payloadcms/payload/blob/main/CONTRIBUTING.md) document in this repository.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] Existing test suite passes locally with my changes

### CODE REVIEW

**Comment by denolfe** [comment]
Will validate this tomorrow.

**Comment by denolfe** [comment]
Just published a beta to test this. The typing is working as expected.

File: packages/payload/src/bin/generateTypes.ts [hunk 1]
```diff
@@ -50,10 +27,8 @@ export async function generateTypes(): Promise<void> {
 
   const jsonSchema = configToJSONSchema(payload.config, payload.db.defaultIDType)
 
-  const collectionDeclaration = generateEntityDeclarations(config.collections, 'collections')
-  const globalDeclaration = generateEntityDeclarations(config.globals, 'globals')
-  const declare = `declare module 'payload' {\n  export interface GeneratedTypes {\n${collectionDeclaration}\n${globalDeclaration}\n  }\n}`
-
+  const declare = `declare module "payload" {\n  export interface GeneratedTypes extends Config {} \n}`
```
Comment (denolfe) [inline packages/payload/src/bin/generateTypes.ts:30]: Small nitpick: remove the space between `{}` and `\n`. This equates to trimming whitespace on the generated file.

File: packages/payload/src/bin/generateTypes.ts [hunk 2]
```diff
@@ -11,29 +11,6 @@ import { configToJSONSchema } from '../utilities/configToJSONSchema'
 import { toWords } from '../utilities/formatLabels'
```
Comment (denolfe) [inline packages/payload/src/bin/generateTypes.ts:11]: Clean up any unused imports.

**Review by denolfe — CHANGES_REQUESTED** [review]
Just some small things. This looks to be functioning as expected, though.

**Comment by xHomu** [comment]
Thanks for the code review!

**Comment by denolfe** [comment]
Thanks for the PR @xHomu !

**Pull request merged**